tests/Api.IntegrationTests/AuthorizationTests.cs

```diff
@@ -59,9 +62,118 @@ public async Task ValidateThatEndpointsHaveProtection()
         }
     }
 
+    [Fact]
+    public async Task ValidateThatMissingApiSecretThrows()
+    {
+        var request = new HttpRequestMessage(HttpMethod.Get, "/credentials/list?userId=1");
+        request.Headers.Add("Accept", "application/json");
+
+        var httpResponse = await _client.SendAsync(request);
+        var body = await httpResponse.Content.ReadAsStringAsync();
+        Assert.Equal(HttpStatusCode.Unauthorized, httpResponse.StatusCode);
+
+        AssertHelper.AssertEqualJson("""
+        {
+            "type": "https://docs.passwordless.dev/guide/errors.html#ApiSecret",
+            "title": "A valid 'ApiSecret' header is required.",
+            "status": 401,
+            "detail": "A valid 'ApiSecret' header is required."
+        }
+        """, body);
+    }
+
+    [Fact]
+    public async Task ValidateThatInvalidApiSecretThrows()
+    {
+        var request = new HttpRequestMessage(HttpMethod.Get, "/credentials/list?userId=1");
+        request.Headers.Add("Accept", "application/json");
+        request.Headers.Add("ApiSecret", _factory.ApiSecret + "invalid");
+        var httpResponse = await _client.SendAsync(request);
+        var body = await httpResponse.Content.ReadAsStringAsync();
+        Assert.Equal(HttpStatusCode.Unauthorized, httpResponse.StatusCode);
+
+        AssertHelper.AssertEqualJson("""
+        {
+            "type": "https://docs.passwordless.dev/guide/errors.html#ApiSecret",
+            "title": "A valid 'ApiSecret' header is required.",
+            "status": 401,
+            "detail": "The value of your 'ApiSecret' is not valid."
+        }
+        """, body);
+    }
+    
+    [Theory]
+    [InlineData("test:public:123", "Your ApiSecret header contained a public ApiKey instead of your 'ApiSecret'.")]
+    [InlineData("verify_123", "A verify token was supplied instead of your 'ApiSecret'.")]
+    [InlineData("register_13", "A register token was supplied instead of your 'ApiSecret'." )]
+    [InlineData("somethingrandom", "We don't recognize the value you supplied for your 'ApiSecret'. It started with: 'somethingr'.")]
+    [InlineData("", "A valid 'ApiSecret' header is required.")]
+    [InlineData("missing", "A valid 'ApiSecret' header is required.")]
+    [InlineData("public-header-instead", "A 'ApiKey' header was supplied when a 'ApiSecret' header should have been supplied.")]
+    public async Task ApiSecretGivesHelpfulAdvice(string input, string details)
+    {
+        var request = new HttpRequestMessage(HttpMethod.Get, "/credentials/list?userId=1");
+        request.Headers.Add("Accept", "application/json");
+
+        if (input == "public-header-instead")
+        {
+            request.Headers.Add("ApiKey", "something");
+        }
+        else if (input != "missing") {

```

Comment:
Javascript style bracing 🥲 , you might need to run `dotnet format` at the root of the repo.